src/giskardpy/plugin_update_constraints.py

```diff
@@ -75,14 +76,12 @@ def parse_constraints(self, cmd):
         :rtype: dict
         """
         for constraint in itertools.chain(cmd.constraints, cmd.joint_constraints, cmd.cartesian_constraints):
-            if constraint.type not in allowed_constraint_names():
-                # TODO test me
-                raise InsolvableException(u'unknown constraint')
             try:
-                C = eval(u'giskardpy.constraints.{}'.format(constraint.type))
-            except NameError as e:
+                C = next(x for x in self.allowed_types if constraint.type == x[0])[1]
+            except StopIteration:
                 # TODO return next best constraint type
```

Comment:
If you want and its easy, you can do this todo as well, instead of just listing all constraint names. but its not important.